src/core/lib/slice/slice_intern.cc

```diff
@@ -152,9 +152,13 @@ grpc_slice grpc_slice_maybe_static_intern(grpc_slice slice,
 }
 
 grpc_slice grpc_slice_intern(grpc_slice slice) {
+  return grpc_slice_intern_internal(slice);
+}
+
+grpc_core::InternalSlice grpc_slice_intern_internal(const grpc_slice& slice) {
```

Comment:
IMHO, this should be the constructor of `InternalSlice`, instead of having yet another separate factory method.

Basically please add ctor to the typed slices, and use the ctors instead.  There some further optimizations that the compiler will do for ctors.

Reply:
Done.